### PR TITLE
WIP: Towards #12251 part 3, deprecate full and remove related exports and docs

### PR DESCRIPTION
**Pull request by Sacha0**
Last step towards JuliaLang/LinearAlgebra.jl#231 (for a description of the overall objective and plan, see https://github.com/JuliaLang/julia/pull/17066#issuecomment-228113687). This pull request deprecates all `full` methods and removes related exports and docs. Thanks and best!

### CODE REVIEW

**Comment by Sacha0** [comment]
Rebased with #17066 and #17079 merged, but added the WIP tag as this pull request needs revision and expansion: Addressing https://github.com/JuliaLang/julia/pull/17066#discussion_r69045577 and https://github.com/JuliaLang/julia/pull/17066#issuecomment-231566180 / https://github.com/JuliaLang/julia/pull/17066#issuecomment-231939146 alongside deprecation of `full` seems advantageous (such that the deprecations can point to replacement functionality). @andreasnoack, what is your current thinking on those fronts? Thanks and best!

File: base/abstractarray.jl [hunk 1]
```diff
@@ -622,8 +622,6 @@ convert{T,S,N}(::Type{AbstractArray{T  }}, A::AbstractArray{S,N}) = convert(Abst
 
 convert{T,N}(::Type{Array}, A::AbstractArray{T,N}) = convert(Array{T,N}, A)
 
-full(x::AbstractArray) = x
```
Comment (jakebolewski) [inline base/abstractarray.jl]: 👏

File: base/deprecated.jl [hunk 2]
```diff
@@ -803,6 +803,32 @@ function transpose(x)
     return x
 end
 
+# Deprecate full in favor of convert. Please see #17066, #17079, and #17082.
```
Comment (tkelman) [inline base/deprecated.jl]: since we're not going to do this now, may as well put it below the 0.6 line

Comment (Sacha0) [inline base/deprecated.jl]: Good catch! Moved deprecations below the 0.6 line. Also preemptively rewrote some `@deprecate`'ions with explicit `depwarns`. Thanks!

**Comment by Sacha0** [comment]
With #12153 on the 1.0 milestone, perhaps this PR should receive a 1.0 tag as well?

**Comment by JeffBezanson** [comment]
Closing as out-of-date.

**Pull request closed without merging**